paddle/fluid/framework/operator.cc

```diff
@@ -944,26 +968,22 @@ void OperatorWithKernel::RunImpl(const Scope& scope,
         }
       }
     }
-  }
 
-  if (FLAGS_check_nan_inf) {
-    for (auto& vname : OutputVars(true)) {
-      auto* var = exec_scope.FindVar(vname);
-      if (var == nullptr) continue;
-      if (var->IsType<framework::LoDTensor>()) {
-        CheckTensorNANOrInf(type_, vname, var->Get<framework::LoDTensor>());
-      } else if (var->IsType<framework::SelectedRows>()) {
-        CheckTensorNANOrInf(type_, vname,
-                            var->Get<framework::SelectedRows>().value());
-      }
+    // To solve issue #15032, have a discussion with @Luotao for cpu inference,
+    // do not cache transfer scope, hence in this case delete transfer scope
+    // after run to avoid memory leak
+    if (transfer_scope && !run_by_executor_ && !enable_cache_transfer_scope_) {
+      scope.DeleteScope(transfer_scope);
+    }
+  } catch (platform::NANINFException exception) {
+    std::rethrow_exception(std::current_exception());
+  } catch (...) {
+    if (FLAGS_fast_check_nan_inf) {
+      // exception sometimes happens in fast_check_nan_inf, which is ok for now.
+      // fix this issue in the next version
+    } else {
+      std::rethrow_exception(std::current_exception());
     }
-  }
-
-  // To solve issue #15032, have a discussion with @Luotao for cpu inference,
-  // do not cache transfer scope, hence in this case delete transfer scope
-  // after run to avoid memory leak
-  if (transfer_scope && !run_by_executor_ && !enable_cache_transfer_scope_) {
-    scope.DeleteScope(transfer_scope);
   }
```

Comment:
Can you narrow down the try... catch range?

Reply:
Is that ok if we do this later? maybe after 1.6 release.